### DISCUSSION TITLE
Can someone clarify start/end for golden hour?

### DISCUSSION THREAD

**Discussion by oskapt** [jensrossbach/node-red-contrib-chronos#38] [Q&A] [answered]
Golden Hour happens in both the morning and the evening. Other nodes that I've worked with define the end of sunrise as the start of golden hour, and the start of golden hour as the hour before sunset. This makes the progression something like:

- sunrise start
- sunrise end (golden hour starts)
- golden hour end
- daytime
- golden hour start
- sunset start (golden hour ends)
- sunset end
- night

This node appears to not follow that structure, because when I set something to trigger at 60 minutes before golden hour start, it never fired in the evening.  Since there is only one set of golden hour start/end, which is in the morning, and which is in the evening? If golden hour start is in the morning, is its end never triggered, and if golden hour end is in the evening, is its start never triggered?

Perhaps the node or the wiki can list its configuration options in chronological order instead of grouping them by a single event's start/stop?

**Comment by jensrossbach** [comment, marked as answer]
Hi, the node does not calculate the times for sunlight phases itself but uses the Node.js module SunCalc instead. The different sunlight times are described 👉🏻[here](https://www.npmjs.com/package/suncalc#sunlight-times). As you can read there, `goldenHourEnd` ("Golden Hour (end)" in the node UI) specifies the time when morning golden hour ends and `goldenHour` ("Golden Hour (start)" in the node UI) specifies the time when evening golden hour starts.

I'm just wondering when you say that the node does not fire at 60 minutes before golden hour start in the evening. I tested it on my side and for _Golden Hour (start)_ the node gave me a time in the evening (based on a location in the middle of Germany). Accordingly when using _Golden Hour (end)_ I got a time in the morning, so it looks good.

I agree with you that it would be better to sort the sunlight phases chronologically in the selection menu. Also it would probably make sense to add a description for the different phases in the wiki. If you like you can create an issue, but I will do this anyway.
